### ISSUE TITLE
Large DataViewJS queries won't render in time

### ISSUE THREAD

**Issue by bppdddqqqq** [oleeskild/obsidian-digital-garden#254]
If the DataViewJS queries are large or time-intensive to process, they won't get parsed into the DOM fast enough for the script to export it as Markdown data. :(

**Comment by bppdddqqqq** [comment]
The issue is most likely related to the fact, that the pipeline for evaluating Javascript data is not truly asynchronous. See Publisher.ts and how the function from `obsidian-dataview` library processes it

**Comment by brandaogabriel7** [comment]
> I seem to encounter the same issue. However, for me, even simple dataviewjs queries are not rendered correctly. An example query:
> 
> ```
> dv.table(["mtime", "link"],
> 	dv.pages('-"templates"')
> 	.where(p => p.file.name === "Python")
> 	.map(p => [
>  		p.file.mtime,
>  		p.file.link
> 	])
> )
> ```
> 
> Published result: <img alt="image" width="350" src="https://user-images.githubusercontent.com/48207708/228440314-e0a4783b-fa56-49e4-93a9-3ef8f76c3633.png">

This is happening to me, too. Any ideas why?

```dataviewjs
dv.list(
    dv.pages().where(p => p['dg-publish'] && p.title)
        .sort(p => p.title)
        .limit(20)
        .map(p => `[[${p.file.name}|${p.title}]]`)
);
```

![image](https://github.com/oleeskild/obsidian-digital-garden/assets/29805002/4e197bc8-cd9d-4bf0-8435-e52d3c630b2b)

**Comment by oleeskild** [comment]
I've just released a new version, that _should_ fix the issues you are having. Let me know if it works after updating or if it's still an issue.

**Comment by zcysxy** [comment]
The dataviewjs renders properly now! However, seems like the links are not written correctly. If the dataviewjs in note A contains a link to note B, the link is pointing to `A/B.md`. Can anyone confirm this?

**Comment by oleeskild** [comment]
Yep you are right. I should have mentioned it here. Links from dataviewjs doesn't work yet. When generating normal dataview queries, we are able to generate markdown, and the plugin can handle linking it correctly. But when running dataviewjs queries, we are actually producing HTML, and the plugin currently doesn't pick up on html links to other notes, and so doesn't process them and point them to the correct URL.   There's an issue tracking it here: #346

**Comment by zcysxy** [comment]
Thanks for the information @oleeskild !

**Comment by oleeskild** [comment]
This has now been fixed in the latest version of the template. Updating the template to version 1.57.0 or later should result in all dataviewjs links behave similar to other links.

**Comment by zcysxy** [comment]
Works perfectly, thanks!! @oleeskild
